routing/router.go

```diff
@@ -1252,6 +1228,107 @@ func (r *ChannelRouter) networkHandler() {
 	}
 }
 
+func (r *ChannelRouter) getMissingBlocks(currentHeight uint32,
+	chainUpdate *chainview.FilteredBlock) {
+
+	outdatedHash, err := r.cfg.Chain.GetBlockHash(int64(currentHeight))
+	if err != nil {
+		log.Error("unable to retrieve outdated block hash")
+	}
+
+	outdatedBlock := &chainntnfs.BlockEpoch{
+		Height: int32(currentHeight),
+		Hash:   outdatedHash,
```

Comment:
If `outdatedHash` is `nil`, there is a possibility of nil-pointer-deref when `RegisterBlockEpochNtfn` calls `GetClientMissedBlocks`.